### PR TITLE
For #9830: Do not request/use WRITE_EXTERNAL_STORAGE on Android 11 and above

### PR DESCRIPTION
**Pull request by czlucius**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by czlucius** [comment]
Here is the issue for this PR: #9830

**Comment by czlucius** [comment]
May I ask, how do I run the test as this is a library and not an app? Thanks.

**Comment by Amejia481** [comment]
Tests were not running because of this issue https://github.com/mozilla-mobile/android-components/pull/10776 as  https://github.com/mozilla-mobile/android-components/commit/d2aa6096020ac551ecd9dda1059d9a4c6131251d landed they should run now :)

**Comment by Amejia481** [comment]
I updated the tests a bit as we can do it as normal JVM tests, I also waiting for https://github.com/mozilla-mobile/android-components/issues/10778 to be addressed as we are not able run tests for this PR until is fixed.

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/manager/FetchDownloadManager.kt
Comment (Amejia481) [inline components/feature/downloads/src/main/java/mozilla/components/feature/downloads/manager/FetchDownloadManager.kt:48]: For we use this download manager that is based of [GeckoView ](https://mozilla.github.io/geckoview/) :)

Comment (czlucius) [reply components/feature/downloads/src/main/java/mozilla/components/feature/downloads/manager/FetchDownloadManager.kt:48]: Ok

**Comment by czlucius** [comment]
> I updated the tests a bit as we can do it as normal JVM tests, I also waiting for #10778 to be addressed as we are not able run tests for this PR until is fixed.
It seems like the tests are working already.

**Comment by Amejia481** [comment]
Not really :(, 
The release engineering team has closed multiple issues but still they are not running. The source of the issue is that your fork repo is renamed [mozilla-android-components](https://github.com/czlucius/mozilla-android-components) instead of the default name [android-components](https://github.com/mozilla-mobile/android-components), I'm not sure if it will be easier to just rename your fork repo and open a new PR. What do you think?

**Comment by Amejia481** [comment]
Now tests are running 🙌🏽

**Comment by Amejia481** [comment]
[As scoped storage is available in Android 10,](https://developer.android.com/training/data-storage) we don't need to require it on Android 10 and above

**Comment by Amejia481** [comment]
I did some manual testing and everything looks good 👍🏽 .
I'll wait until Monday to land this patch, as we avoid landing features over the weekend.

**Comment by czlucius** [comment]
> Not really :(, 
> The release engineering team has closed multiple issues but still they are not running. The source of the issue is that your fork repo is renamed [mozilla-android-components](https://github.com/czlucius/mozilla-android-components) instead of the default name [android-components](https://github.com/mozilla-mobile/android-components), I'm not sure if it will be easier to just rename your fork repo and open a new PR. What do you think? 

I can rename it back if necessary

**Comment by czlucius** [comment]
> I did some manual testing and everything looks good 👍🏽 .
> I'll wait until Monday to land this patch, as we avoid landing features over the weekend.

Thanks

**Comment by czlucius** [comment]
Do the list fixes to the code still need to be made?

**Review by Amejia481 — APPROVED** [review]

**Comment by Amejia481** [comment]
> Do the list fixes to the code still need to be made?

Thanks for asking, they are already addressed. The patch is on is way to be merged :)

**Comment by Amejia481** [comment]
Thanks @czlucius  for all you effort 🥇 !

**Comment by czlucius** [comment]
> Do the list fixes to the code still need to be made?

Sorry, I made a typo. I meant "lint" fixes

**Comment by Amejia481** [comment]
> > Do the list fixes to the code still need to be made?
> 
> Sorry, I made a typo. I meant "lint" fixes

No worries, they are also addressed :)

**Pull request merged**
